### ISSUE TITLE
Data pack not showing races

### ISSUE THREAD

**Issue by BadmanTru** [my-name-is-samael/BeamJoy#63]
I extracted everything to senrii and loaded into west coast and no races show loaded. It would be nice to get documentation.

**Comment by BadmanTru** [comment]
also is it possible to spawn custom configs for races?

**Comment by my-name-is-samael** [comment]
1: Did you restart your server if you were already on west coast ?
Data can be hotloaded only if you switch map or restart the server

2: Of course you can spawn custom configs for races and even enforce a custom config for all players

**Comment by BadmanTru** [comment]
Yes server was restarting, I'd go to races and it shows 0/10.

**Comment by my-name-is-samael** [comment]
Do you have all the datapack JSON files in the `scenarii` folder ?

**Comment by BadmanTru** [comment]
yes i do, that is correct.

**Comment by BadmanTru** [comment]
This is how I have it in the server 
![Untitled2](https://github.com/user-attachments/assets/654acbde-ee99-45d1-9c03-9fcbb869c343)

it doesn't allow me to unarchive it
but everything works aside from the data pack

- I unarchived and installed the data pack, then repacked it and reuploaded it to the server.

**Comment by my-name-is-samael** [comment]
This is not the right place, they should be placed under `Resources/Server/BeamJoyData/db/scenarii`

**Comment by BadmanTru** [comment]
ah, thank you! 
I will fix that tonight. 

I appreciate the quick feedback. 

I did mention previously on the beammp resources page of a issue with performance that happens. 

It seems anytime I open the "settings" tab, it causes fps to go down to 5fps. I will get a screen record for you tonight.

**Comment by my-name-is-samael** [comment]
I'll let you close the issue when you will try out and witness everything working fine.

I remember your comment but it is really out of context to tell something is caused by constant tick update, videogames are only working because of constant tick updates.
I know sub-windows are pretty laggy (the more there is content to render inside though), but it is BeamNG's choice to implement IMGUI (for world editor). I tended to limit windows amount for players, and admins should only use them to setup things, not to play along. I thought about creating UI Apps as mod interface but it is far worse because players can remove them, causing the mod to become pointless and informations loss on the way, and I don't want to force an UI App, it is too intrusive and it completely breaks UX.
There is no good choice in this case, I just did my best.

**Comment by BadmanTru** [comment]
You are correct, I'm very limited knowledge on this subject. I'll do my best to correct that in the future of reporting. 

I agree, i think the ui is perfectly fine as it is. (and you are correct, it's not really needed during regular usage to go into the sub ui menus which is when it lags) so not to big of a issue. 

I really do appreciate the amount of hard work you put into this. It's just wonderful.

**Comment by BadmanTru** [comment]
we keep getting crashes 

![image](https://github.com/user-attachments/assets/67386045-a194-4d45-b55c-042e878f0086)

**Comment by my-name-is-samael** [comment]
I don't see any crash on your screenshot

**Comment by BadmanTru** [comment]
![image](https://github.com/user-attachments/assets/0891a0b1-9f82-410a-b9b8-275d2bde849b)

one guy keeps getting this error

**Comment by my-name-is-samael** [comment]
This will need further investigation.

Is your guy launching in vulkan mode (directx mode is now prohibited since it is extremely instable now) ?
Does this crash occurring on a specific action/combination of actions ?

**Comment by BadmanTru** [comment]
So he was sitting in the server just fine unspawned, when I spawned in is when he crashed. If no cars on track it doesn't kick him. 

He joins in, sits for 5 seconds (when car is spawned) and gives me a error code similar to that crash code i posted.

He is the only one having this issue, hes done a full reinstall of both beam, beammp, I had him clear out the mp resources folder, and verify game data to. 

No luck

**Comment by my-name-is-samael** [comment]
Does he launch with Vulkan mode ?
If he is the only one having the issue, it is not coming from the mod, his computer specs must conflict with the game itself or BeamMP.

He can try to launch solo, spawn his vehicle, then spawn traffic to see if the crash is happening again (to determine if it is coming from the game or BeamMP).

**Comment by BadmanTru** [comment]
He has not been launching in vulkan mode.
